src/TestingDevice.java

```diff
@@ -1,7 +1,24 @@
 import java.util.ArrayList;
```

Comment:
General comment, for this file there is a lot of if/else if statements without brackets. I know it's your code style but it's easier to read with the brackets. Since this isn't going to be turned in this will be a soft change request - totally up to you. But future for future group projects it's better to put the brackets in to make it more readable and less error prone.

Reply:
I agree with you.
This was a poorly-reviewed commit on my part.

I usually ammend bad habits like this. Good eye.

I have implemented this requested change in the bugfixes branch.